src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,143 @@
+            {
+                // Check that the object is a StringBuilder
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                IMethodSymbol appendStringMethod = stringBuilderType
+                    .GetMembers("Append")
+                    .OfType<IMethodSymbol>()
+                    .WhereAsArray(s =>
+                        s.Parameters.Length == 1 &&
+                        s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                    .FirstOrDefault();
+                if (appendStringMethod is null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    IInvocationOperation invocationOperation = (IInvocationOperation)operationContext.Operation;
+                    if (invocationOperation.Arguments.Length < 1)
+                    {
+                        return;
+                    }
+
+
+                    if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+                    IArgumentOperation firstArgument = arguments[0];
+
+                    // We don't handle class fields. Only local declarations
+                    IOperation firstArgumentValue = firstArgument.Value;
+                    if (firstArgumentValue is ILocalReferenceOperation argumentValue)
+                    {
+                        HandleLocalReferenceOperation(argumentValue, operationContext);
+                        return;
+                    }
+                    else if (firstArgumentValue is ILiteralOperation literalArgumentValue)
+                    {
+                        HandleLiteralReferenceOpearion(literalArgumentValue, operationContext);
+                        return;
+                    }
+                    return;
+                }
+                , OperationKind.Invocation);
+            });
+        }
+
+        private static void HandleLiteralReferenceOpearion(ILiteralOperation literalArgumentValue, OperationAnalysisContext operationContext)
+        {
+            if (literalArgumentValue.ConstantValue.HasValue && literalArgumentValue.ConstantValue.Value is string unitString && unitString.Length == 1)
+            {
+                operationContext.ReportDiagnostic(literalArgumentValue.CreateDiagnostic(Rule));
+            }
+        }
+
+        private static void HandleLocalReferenceOperation(ILocalReferenceOperation argumentValue, OperationAnalysisContext operationContext)
+        {
+            ILocalSymbol localArgumentDeclaration = argumentValue.Local;
```

Comment:
This is not the declaration site for the local.

```suggestion
            ILocalSymbol localArgumentOperation = argumentValue.Local;
```